scapy/layers/inet6.py

```diff
@@ -2138,7 +2138,8 @@ class NonceField(StrFixedLenField):
     def __init__(self, name, default=None):
         StrFixedLenField.__init__(self, name, default, 8)
         if default is None:
-            self.default = self.randval()
+            self.default = struct.pack("II", random.randint(0, 4294967295),
```

Comment:
Unfortunately, I spent a lot of time trying to come up with a better fix. Sometimes on Python3, the nonce is longer than 8 bytes with the current code...